src/core/toga/sources/base.py

```diff
@@ -0,0 +1,71 @@
+
+
+class Datum:
+    def __init__(self, value=None, icon=None):
+        self.value = value
+        self.icon = icon
```

Comment:
Do you see this being expandable? To put another way, is icon the only "special field" we imagine existing? If not, should this be instead a `**kwargs` situation?

Reply:
I would expect end users to provide their own Datum (or equivalent). "value" and "icon" are defined because they're used by TogaCell, and we use Datum internally to set up data for the TogaCell.